### ISSUE TITLE
Using `S(variable)` put translations in not used anymore section

### ISSUE THREAD

**Issue by louisroyer** [FaceDeer/update_translations#8]
Add a syntax in `.tr` files to let the user say some translations are associated with a `S(variable)` call in a file. Since it is not trivial to detect variable content unless we run the script (and this is a bad idea), this kind of translations needs to be reviewed manually and not be confused with not used anymore translation. When detection of a `S(variable)` in a file occurs but no string is associated to the variable yet, the script could emit a warning and write a warning in the "not used anymore" section.

**Comment by FaceDeer** [comment]
To be clear, you're talking about a scenario like this?

```
local baseText = "this string should be translated"
local translatedText = S(baseText)
```

And then manually adding

```
#actuallyUsedTag
this string should be translated=
```

To the template and .tr files with this script recognizing #actuallyUsedTag (or whatever the tag turns out to be) and handling the line specially to keep it out of the not-used-anymore section?

Seems reasonable to me.

**Comment by louisroyer** [comment]
Yes, this is a scenario like that (see https://github.com/minetest-mods/moreblocks/commit/86502f295ac5e622851a0d5afb9b82b307989907#diff-673c28fb12d42fc2219aa4567e917ddb for my use case).

**Comment by Wuzzy2** [comment]
The solution is to add a no-op function `local NS = function(s) return s end` at the top of your file and then mark the string literal you want to become translatable with it.

The above code could then be fixed like this:

```
local NS = function(s) return s end
local baseText = NS("this string should be translated")
local translatedText = S(baseText)
```

This works because NS just returns its argument but will be detected by the script. I have documented this my fork of the script: https://codeberg.org/Wuzzy/minetest_update_translations
